### PR TITLE
fix layer_info and fp16 for multi-target multi-output

### PR DESCRIPTION
**Pull request by mnpinto**

### CODE REVIEW

File: fastai2/callback/hook.py [hunk 1]
```diff
@@ -146,7 +146,7 @@ def layer_info(learn):
     def _track(m, i, o):
         return (m.__class__.__name__,)+total_params(m)+(apply(lambda x:x.shape, o),)
     layers = [m for m in flatten_model(learn.model)]
-    xb,_ = learn.dbunch.train_dl.one_batch()
+    xb,*_ = learn.dbunch.train_dl.one_batch()
```
Comment (sgugger) [inline fastai2/callback/hook.py]: Batches are always a tuple (of maybe nested tuples) so this should not be necessary.

File: fastai2/callback/fp16.py [hunk 2]
```diff
@@ -89,7 +89,12 @@ def begin_fit(self):
         if self.dynamic: self.count = 0
 
     def begin_batch(self): self.learn.xb = to_half(self.xb)
-    def after_pred(self):  self.learn.pred = self.pred.float()
+
+    def after_pred(self):
+        if isinstance(self.learn.pred, list):
```
Comment (sgugger) [inline fastai2/callback/fp16.py]: There is the to_float method to do this normally.

File: fastai2/callback/hook.py (hunk 1 above)
Comment (mnpinto) [reply fastai2/callback/hook.py]: If the tuple len > 2 it gives this error:
image

File: fastai2/callback/fp16.py (hunk 2 above)
Comment (mnpinto) [reply fastai2/callback/fp16.py]: Thanks, I will update it

File: fastai2/callback/hook.py (hunk 1 above)
Comment (sgugger) [reply fastai2/callback/hook.py]: Oh, it's actually something different, we need to cut the tuple at the right place. Let's merge this part as is and I'll follow up to add that part.

**Comment by sgugger** [comment]
Thanks! You can check on the commit I referenced the correct code for grabbing the right xb.

**Pull request merged**